From d82cfc6f1677e7c3928f0c5512f592ab9b62ad06 Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Sun, 11 Mar 2012 15:47:00 +0200
Subject: [PATCH 18/18] mmc: omap_hsmmc: set pm_runtime_irq_safe

Declare the pm runtime handlers as safe to run
in atomic context (it's in fact being called
from mmc_signal_sdio_irq() in irq context)

Signed-off-by: Eliad Peller <eliad@wizery.com>
---
 drivers/mmc/host/omap_hsmmc.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/mmc/host/omap_hsmmc.c b/drivers/mmc/host/omap_hsmmc.c
index 736dd20..674e60f 100644
--- a/drivers/mmc/host/omap_hsmmc.c
+++ b/drivers/mmc/host/omap_hsmmc.c
@@ -2473,6 +2473,7 @@ static int __init omap_hsmmc_probe(struct platform_device *pdev)
 	host->dpm_state = DISABLED;
 
 	pm_runtime_enable(host->dev);
+	pm_runtime_irq_safe(host->dev);
 #ifndef CONFIG_PM_RUNTIME
 	/*
 	 * If runtime PM is not enabled, ensure clocks are always enabled.
-- 
1.7.6.401.g6a319

